```{index} Force method; for truss structures
```

# ... for truss structures

The general concept of the force method is shown on [](./force_method.md) and covered in chapter 2.1 while the force method for truss structures is treated in in chapter 2.2.8 - 2.2.10 of the book Mechanica, Statisch onbepaalde constructies en bezwijkanalyse (in Dutch) {cite:p}`Hartsuijker2016`.

We'll cover the application to truss structures with the following example

::::::{prf:example}
:nonumber: true
:label: sd_ext_0

```{figure} ./extension_data/Example.svg
---
name: example_sd
align: center
---
Example structure
```

::::::

1. Determine [the degree of statical determinacy](../determinancy.md).

::::::{prf:example}
:nonumber: true
:label: sd_ext_1

For our example, we might be interested in the internal force distribution, so we need to evaluate the degree of internal statical determinacy.

```{figure} ./extension_data/unknown_forces.svg
---
name: example_sd_1
align: center
---
There are 18 unknown forces.
```

```{figure} ./extension_data/equations.svg
---
name: example_sd_1
align: center
---
There are 17 equilibrium equations
```

So this structure is 1st order internally statically indeterminant.

::::::

2. Transform the structure in a statical determinant system by releasing support or adding hinges: add unknown statically indeterminate forces and displacement constraints for each of the support you released and hinges you added.

::::::{prf:example}
:nonumber: true
:label: sd_ext_2

There are many options here, of which a few are shown below:

```{figure} ./extension_data/options.svg
---
name: example_sd_1
align: center
---
A few options to transform the structure in a statical determinant system
```

The last option is chosen.

::::::

3. Solve for the displacement in terms of the unknown indeterminate forces as you would normally do for a statically determinate structure.
4. Use your displacement constraints to solve for the statically indeterminate forces

In [67]:
import sympy as sym

L_1, L_2, L_3, q, N, Bv = sym.symbols('L_1 L_2 L_3 q N Bv')

EA = sym.symbols('EA')

L_1 = sym.nsimplify(4)
L_2 = sym.nsimplify(4)
L_3 = sym.nsimplify(5)
q = sym.nsimplify(28)
display(q)
eq1 = sym.Eq(Bv * (L_1 + L_2) - q * L_2 * (L_1 + L_2/2 ) + N * 4 / 5 * L_1 , 0)
N = sym.solve(eq1, N)[0]
display(N)
L_CD = L_1 / 3 * 5
delta_L = N * L_CD / EA
display(delta_L)
w_D = delta_L / 4 * 5
display(w_D.evalf())
w_E = w_D / L_1 * (L_1 + L_2)
display(w_E.evalf())
delta_L_BE = Bv * L_3 / EA
w_B = w_E - delta_L_BE
display(w_B)
display(w_B.simplify().evalf())
eq2 = sym.Eq(w_B, 0)
B_v_sol = sym.solve(eq2, Bv)[0]
display(B_v_sol)
display(N.subs(Bv, B_v_sol))

28

210 - 5*Bv/2

(1400 - 50*Bv/3)/EA

1.25*(1400.0 - 16.6666666666667*Bv)/EA

2.5*(1400.0 - 16.6666666666667*Bv)/EA

-5*Bv/EA + 5*(1400 - 50*Bv/3)/(2*EA)

46.6666666666667*(75.0 - Bv)/EA

75

45/2

In [57]:
import sympy as sym

L_1, L_2, L_3, q, N, Bv = sym.symbols('L_1 L_2 L_3 q N Bv')

EA = sym.symbols('EA')

L_1 = sym.nsimplify(3)
L_2 = sym.nsimplify(4)
L_3 = sym.nsimplify(5)
q = sym.nsimplify(32)
display(q)
eq1 = sym.Eq(Bv * (L_1 + L_2) - q * L_2 * (L_1 + L_2/2 ) + N * L_1 , 0)
N = sym.solve(eq1, N)[0]
display(N)
L_CD = L_1
delta_L = N * L_CD / EA
display(delta_L)
w_D = delta_L
display(w_D.evalf())
w_E = w_D / L_1 * (L_1 + L_2)
display(w_E.evalf())
delta_L_BE = Bv * L_3 / EA
w_B = w_E - delta_L_BE
display(w_B)
eq2 = sym.Eq(w_B, 0)
B_v_sol = sym.solve(eq2, Bv)[0]
display(B_v_sol)
display(N.subs(Bv, B_v_sol))

32

640/3 - 7*Bv/3

(640 - 7*Bv)/EA

(640.0 - 7.0*Bv)/EA

2.33333333333333*(640.0 - 7.0*Bv)/EA

-5*Bv/EA + 7*(640 - 7*Bv)/(3*EA)

70

50

In [33]:
2.5*16.6666667

41.666666750000005

In [22]:
50/3*5/4*2

41.66666666666667

## Exercises
- Exercises 2.31 - 2.41, in chapter 2.3 of the book Mechanica, Statisch onbepaalde constructies en bezwijkanalyse (in Dutch) {cite:p}`Hartsuijker2016`.
Answers are available on [this website](https://icozct.tudelft.nl/TUD_CT/boekantwoorden/vol3/Chapter1-2/).